lib/psych/visitors/to_ruby.rb

```diff
@@ -371,9 +371,8 @@ def revive_hash hash, o
       if String.method_defined?(:-@)
         def deduplicate key
           if key.is_a?(String)
-            # It is important to untaint the string, otherwise it won't
-            # be deduplicated into and fstring, but simply frozen.
-            -(key.untaint)
+            key.untaint if RUBY_VERSION < '2.7'
```

Comment:
Could we make this check "static" (like `if String.method_defined?(:-@)`). `deduplicate` is quite an hot spot, and I'm worried about the performance impact of an extra string comparison.

Reply:
Sure, this makes sense.